core/blockchain/upgradedb.go

```diff
@@ -2,61 +2,13 @@ package blockchain
 
 import (
 	"fmt"
-	"github.com/Qitmeer/qng/common/roughtime"
-	"github.com/Qitmeer/qng/core/dbnamespace"
-	"github.com/Qitmeer/qng/core/serialization"
-	"github.com/Qitmeer/qng/database/common"
-	"github.com/Qitmeer/qng/database/legacydb"
-	l "github.com/Qitmeer/qng/log"
 )
 
 // update db to new version
 func (b *BlockChain) upgradeDB(interrupt <-chan struct{}) error {
-	version8 := uint32(8)
-	version9 := uint32(9)
-	version10 := uint32(10)
-	version11 := uint32(11)
 	if b.dbInfo.Version() == currentDatabaseVersion {
 		return nil
-	} else if b.dbInfo.Version() == version8 ||
-		b.dbInfo.Version() == version9 ||
-		b.dbInfo.Version() == version10 ||
-		b.dbInfo.Version() == version11 {
-		return fmt.Errorf("Your database version(%d) is too old and can only use old qng (release-v1.0.20)\n", b.dbInfo.Version())
+	} else {
+		return fmt.Errorf("Your database version(%d) is too old and can only use old qng\n", b.dbInfo.Version())
```

Comment:
before this, I think better to have a fixed version to the latest old compatible qng.

Reply:
[3823c6f](https://github.com/Qitmeer/qng/pull/502/commits/3823c6ff21e8cc2599da97ecd5b0adf14893de02)